python/lsst/ap/association/transformDiaSourceCatalog.py

```diff
@@ -0,0 +1,234 @@
+        diaSourceDf["ccdVisitId"] = ccdVisitId
+        diaSourceDf["filterName"] = band
+        diaSourceDf["midPointTai"] = diffIm.getInfo().getVisitInfo().getDate().get(system=DateTime.MJD)
+        diaSourceDf["diaObjectId"] = 0
+        diaSourceDf["htmId20"] = 0
+
+        df = self.transform(band,
+                            ParquetTable(dataFrame=diaSourceDf),
+                            self.funcs,
+                            dataId=None).df
+        return pipeBase.Struct(
+            diaSourceTable=df
+        )
+
+    def computeBBoxSizes(self, inputCatalog):
+        """Compute the size of a square bbox that fully contains the detection
+        footprint.
+
+        Parameters
+        ----------
+        inputCatalog : `lsst.afw.table.SourceCatalog`
+            Catalog containing detected footprints.
+
+        Returns
+        -------
+        outputBBoxSizes : `numpy.ndarray`, (N,)
+            Array of bbox sizes.
+        """
+        outputBBoxSizes = np.empty(len(inputCatalog), dtype=int)
+        for idx, record in enumerate(inputCatalog):
+            footprintBBox = record.getFootprint().getBBox()
+            # Compute twice the size of the largest dimension of the footprint
+            # bounding box. This is the largest footprint we should need to cover
+            # the complete DiaSource assuming the centroid is withing the bounding
+            # box.
+            maxSize = 2 * np.max([footprintBBox.getWidth(),
+                                  footprintBBox.getHeight()])
+            recX = record.getCentroid().x
+            recY = record.getCentroid().y
+            bboxSize = int(
+                np.ceil(2 * np.max(np.fabs([footprintBBox.maxX - recX,
+                                            footprintBBox.minX - recX,
+                                            footprintBBox.maxY - recY,
+                                            footprintBBox.minY - recY]))))
+            if bboxSize > maxSize:
+                bboxSize = maxSize
+            outputBBoxSizes[idx] = bboxSize
+
+        return outputBBoxSizes
+
+    def transform(self, band, parq, funcs, dataId):
+        analysis = self.getAnalysis(parq, funcs=funcs, band=band)
+        df = analysis.df
+        if dataId is not None:
+            for key, value in dataId.items():
+                df[key] = value
+
+        return pipeBase.Struct(
+            df=df,
+            analysis=analysis
+        )
+
+    def getAnalysis(self, parq, funcs=None, band=None):
+        # Avoids disk access if funcs is passed
+        if funcs is None:
+            funcs = self.getFunctors()
+        analysis = PostprocessAnalysis(parq, funcs, filt=band)
+        return analysis
+
+    def getFunctors(self):
+        funcs = CompositeFunctor.from_file(self.config.functorFile)
+        funcs.update(dict(PostprocessAnalysis._defaultFuncs))
+        return funcs
+
+    # Below are temporary functions to preserve functionality before
```

Comment:
I'm assuming this comment means the methods below are used in some other code somewhere for the time being?  If not, they should probably be removed.

Reply:
Correct. You just caught me before I got rid of them. Check the compare again and you'll see that this and `getFunctors` is now gone and inherited from the `TransformBaseCatalogTask` in postprocess.py in `pipe_tasks`.